tests/extendHTMLElement.spec.ts

```diff
@@ -0,0 +1,70 @@
+import Component from '../src';
+
+describe('#Extending HTMLElements', (): void => {
+  // register an extended set on HTMLElements
+  beforeEach((): void => {
+    // eslint-disable-next-line global-require
+    const basicHTML = require('basichtml');
+    // eslint-disable-next-line global-require
+    const basicHTMLElements = require('basichtml-elements');
+    const customElements = new basicHTML.CustomElementRegistry();
+
+    // define things
+    basicHTMLElements.defineOnRegistry(customElements);
+
+    basicHTML.init({
+      window: global,
+      customElements,
+    });
+  });
+
+  it('extends non builtin element and check against HTMLElement', (): void => {
+    class TestComponent extends Component<object, object> {
+      public static componentName = 'test-component';
+    }
+
+    TestComponent.register();
+
+    const element = new TestComponent();
+    expect(element).toBeInstanceOf(HTMLElement);
+  });
+
+  it('extends a button element and check against HTMLElement', (): void => {
+    class TestButton extends Component<object, object> {
+      public static componentName = 'test-button';
+
+      public static basedOn = 'button';
+    }
+
+    TestButton.register();
+
+    const element = new TestButton();
+    expect(element).toBeInstanceOf(HTMLElement);
+  });
+
+  it('extends a button element and check against default button', (): void => {
+    class TestButton extends Component<object, object> {
+      public static componentName = 'test-button';
+
+      public static basedOn = 'button';
+    }
+
+    TestButton.register();
+
+    const element = new (customElements.get('test-button'))();
+    expect(element).toBeInstanceOf(customElements.get('button'));
+  });
+
+  it('extends a button element and check against input field', (): void => {
```

Comment:
If `a=1`, we can test if `a` is `1` (this is done in L45). This test (L58) is testing that, under the same conditions of L45, `a` is not `2`. Mathematically, I don't think this test is useful (i.e. if `1` is not `2`, and `a=1`, by default, `a!=2`).

IMO would be better to test:
set `a=1` => test if `a` is `1`
set `a=2` => test if `a` is `2`

(`a`= Component, `1`= button, `2` = input)

Reply:
I made that test to be sure that it does not return true on every type of element...

Reply:
Yep, I inferred the intention. I'm just saying the end result (the test on the PR) is something that will always return true if we don't hardcode "button" as the `baseOn` attribute.
The test I propose would do the exact same thing, without the mathematical fallacy.

Reply:
changed it the way you suggest it.